src/components/DataTableSkeleton/DataTableSkeleton.js

```diff
@@ -27,6 +27,16 @@ const DataTableSkeleton = ({
     [`${prefix}--data-table-v2--compact`]: compact,
   });
 
+  let normalizedHeaders;
+
+  if (typeof headers[0] === 'object') {
```

Comment:
I think this type check would yield false positives (if `headers[0]` was `null` for example), would we want to use something like `headers[0] === Object(headers[0]) && !Array.isArray(headers[0])`?

Reply:
Good point -- IMO it never hurts to be extra safe here 👍